pyfolio/utils.py

```diff
@@ -259,3 +260,45 @@ def extract_rets_pos_txn_from_zipline(backtest):
     transactions.index.tz = None
 
     return returns, positions, transactions, gross_lev
+
+
+SETTINGS = {
+    # Settings dict to store functions/values that may
+    # need to be overridden depending on the users environment
+    'returns_func': default_returns_func
+}
+
+
+def register_return_func(func):
+    """
+    Registers the function that will be called for
+    retrieving returns data.
+
+    Parameters
+    ----------
+    func : function
+        A function that accepts a symbol and
+        returns a pandas.Series of asset returns.
+
+    Returns
+    -------
+    None
+    """
+    SETTINGS['returns_func'] = func
+
+
+def get_symbol_rets(symbol):
```

Comment:
@humdings @twiecki how would you thread those additional arguments to `get_symbol_rets`.  The whole point of this function is that it's being called behind the scenes by `create_*_tearsheet` with a symbol.  If you wanted to forward arguments, then all the tearsheet functions would need to accept additional args and kwargs, at which point you might as well just load the benchmark explicitly as an array and pass it in.

If your symbol returns function needs database credentials, it can close over them like this:

```
creds = get_db_credentials()
connection = DatabaseConnection(creds)

def my_benchmark_func(symbol):
    return connection.make_some_query(...)
```

This is also a perfectly good use-case for `functools.partial`.